client/src/app/pages/applications/applications-table/components/manage-columns-modal.tsx

```diff
@@ -0,0 +1,103 @@
+import "./manage-columns-modal.css";
+
+import React, { useState } from "react";
+import {
+  Button,
+  DataList,
+  DataListCell,
+  DataListCheck,
+  DataListControl,
+  DataListItem,
+  DataListItemCells,
+  DataListItemRow,
+  Modal,
+  Text,
+  TextContent,
+  TextVariants,
+} from "@patternfly/react-core";
+import { ColumnState } from "@app/hooks/table-controls/column/useColumnState";
+
+export interface ManagedColumnsProps<TColumnKey extends string> {
+  onClose(): void;
+  columns: ColumnState<TColumnKey>[];
+  setColumns: (newColumns: ColumnState<TColumnKey>[]) => void;
+  description?: string;
+  saveLabel?: string;
+  cancelLabel?: string;
+  title?: string;
+}
+
+export const ManageColumnsModal = <TColumnKey extends string>({
+  description = "Selected columns will be displayed in the table.",
+  onClose,
+  columns,
+  setColumns,
+  saveLabel = "Save",
+  cancelLabel = "Cancel",
+  title = "Manage Columns",
+}: ManagedColumnsProps<TColumnKey>) => {
+  const [editedColumns, setEditedColumns] =
+    useState<ColumnState<TColumnKey>[]>(columns);
+
+  const onSelect = (id: TColumnKey, isVisible: boolean): void => {
+    setEditedColumns(
+      editedColumns.map((col) => ({
+        ...col,
+        isVisible: col.id === id ? isVisible : col.isVisible,
+      }))
+    );
+  };
+
+  const onSave = () => {
+    // If ordering is implemented, update accordingly
+    setColumns(editedColumns);
+    onClose();
+  };
+
+  return (
+    <Modal
+      title={title}
+      isOpen={true}
+      variant="small"
+      description={
+        <TextContent>
+          <Text component={TextVariants.p}>{description}</Text>
+        </TextContent>
+      }
+      onClose={onClose}
+      actions={[
+        <Button key="save" variant="primary" onClick={onSave}>
+          {saveLabel}
+        </Button>,
+        <Button key="cancel" variant="secondary" onClick={onClose}>
+          {cancelLabel}
+        </Button>,
+      ]}
+    >
+      <DataList aria-label={title} id="table-column-management" isCompact>
+        {editedColumns.map(({ id, label, isVisible }, index) => (
+          <DataListItem key={index}>
+            <DataListItemRow className="custom-data-list-item-row">
+              <DataListControl>
+                <DataListCheck
+                  aria-labelledby={`check-${id}`}
+                  checked={isVisible}
+                  isDisabled={id === "name"}
```

Comment:
Humm....this feels a bit awkward.

I'd rather see a flag of some kind be passed along with the columns as a way to decided what columns are required.  That could be left for a future enhancement as long as some comments are added along the lines of "any column whose name is 'name' will not be allowed to be hidden".